pkg/odo/cli/pipelines/bootstrap.go

```diff
@@ -85,5 +92,7 @@ func NewCmdBootstrap(name, fullName string) *cobra.Command {
 	}
 
 	bootstrapCmd.Flags().StringVarP(&o.prefix, "prefix", "p", "", "add a prefix to the environment names")
+	bootstrapCmd.Flags().StringVarP(&o.tokenFilename, "github-token", "", bootstrap.DefaultTokenFilename, "filename for GitHub auth token")
+	bootstrapCmd.Flags().StringVarP(&o.quayIOAuthFilename, "quay-io-auth-json", "", bootstrap.DefaultQuayIOAuthFilename, "filename for quay IO auth json")
 	return bootstrapCmd
```

Comment:
This option should be "quay-auth-filename to the user.    I am not 100% sure it should be called "quay".  It is really a docker thing.

Reply:
I'd like to see some testing of the required parameters...one way would be to do something like:

```go
func TestBootstrapCommandWithMissingParams(t *testing.T) {
       cmdTests := []struct {
               args    []string
               wantErr string
       }{
               {[]string{}, `Required flag(s) "base-repository", "dockerconfigjson", "github-token", "quay-username" have/has not been set`},
       }

       for _, tt := range cmdTests {
               cmd := NewCmdBootstrap(BootstrapRecommendedCommandName, util.GetFullName("odo", BootstrapRecommendedCommandName))
               cmd, _, err := executeCommand(cmd)

               if err.Error() != tt.wantErr {
                       t.Errorf("got %s, want %s", err, tt.wantErr)
               }
       }
}
```
This should be fleshed out a bit with other combinations and options.

With a function that executes a cobra Command:

```go
func executeCommand(cmd *cobra.Command, args ...string) (c *cobra.Command, output string, err error) {
       buf := new(bytes.Buffer)
       cmd.SetOutput(buf)
       cmd.SetArgs(args)
       c, err = cmd.ExecuteC()
       return c, buf.String(), err
}
```

Reply:
Added the test case, kindly review. :)